keras_cv/models/object_detection/yolox/binary_crossentropy.py

```diff
@@ -0,0 +1,85 @@
+# Copyright 2023 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import tensorflow as tf
+
+
+class BinaryCrossentropy(tf.keras.losses.Loss):
+    """Computes the cross-entropy loss between true labels and predicted labels.
+    Use this cross-entropy loss for binary (0 or 1) classification applications.
+    This loss is updated for YoloX by offering support for no axis to mean over.
+    Args:
+        from_logits: Whether to interpret `y_pred` as a tensor of
+            [logit](https://en.wikipedia.org/wiki/Logit) values. By default, we
+            assume that `y_pred` contains probabilities (i.e., values in [0,
+            1]).
+        label_smoothing: Float in [0, 1]. When 0, no smoothing occurs. When >
+            0, we compute the loss between the predicted labels and a smoothed
+            version of the true labels, where the smoothing squeezes the labels
+            towards 0.5.  Larger values of `label_smoothing` correspond to
+            heavier smoothing.
+        axis: the axis along which to mean the ious. Defaults to `None` which implies
+            mean across no axes.
+    Sample Usage:
+    ```python
+    model.compile(
+      loss=keras_cv.models.object_detection.yolox.binary_crossentropy.BinaryCrossentropy(from_logits=True)
+      ....
+    )
+    ```
+    """
+
+    def __init__(
+        self, from_logits=False, label_smoothing=0.0, axis=None, **kwargs
+    ):
+        super().__init__(**kwargs)
+        self.from_logits = from_logits
+        self.label_smoothing = label_smoothing
+        self.axis = axis
+
+    def call(self, y_true, y_pred):
+        y_pred = tf.convert_to_tensor(y_pred)
+        y_true = tf.cast(y_true, y_pred.dtype)
+        label_smoothing = tf.convert_to_tensor(
+            self.label_smoothing, dtype=y_pred.dtype
+        )
+
+        def _smooth_labels():
+            return y_true * (1.0 - label_smoothing) + 0.5 * label_smoothing
+
+        y_true = tf.__internal__.smart_cond.smart_cond(
+            label_smoothing, _smooth_labels, lambda: y_true
```

Comment:
do we need to do label_smoothing > 0 for this?

Reply:
That's what this condition intends to do. I believe this is equivalent to `label_smoothing > 0`?

Either way, this is a direct copy from the [Keras implementation](https://github.com/keras-team/keras/blob/v2.12.0/keras/losses.py#L2151). Would it make more sense to add any changes to Keras and then I can replicate them here?